File: PrismaApi/PrismaApi.Domain/Dtos/DecisionDtos.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace PrismaApi.Domain.Dtos;

public class DecisionDto
{
    [JsonPropertyName("id")]
    public Guid Id { get; set; } = Guid.NewGuid();
    [JsonPropertyName("issue_id")]
    public Guid IssueId { get; set; }
}

public class DecisionIncomingDto : DecisionDto
{
    [JsonPropertyName("options")]
    public List<OptionIncomingDto> Options { get; set; } = new();
    [JsonPropertyName("type")]
    public string Type { get; set; } = "Focus";
}

public class DecisionOutgoingDto : DecisionDto
{
    [JsonPropertyName("options")]
    public List<OptionOutgoingDto> Options { get; set; } = new();
    [JsonPropertyName("type")]
    public string Type { get; set; } = "Focus";
}

File: PrismaFastApi/src/dtos/decision_dtos.py
import uuid
from pydantic import BaseModel, Field
from typing import List
from src.dtos.option_dtos import (
    OptionIncomingDto,
    OptionOutgoingDto,
)
from src.constants import DecisionHierarchy


class DecisionDto(BaseModel):
    id: uuid.UUID = Field(default_factory=uuid.uuid4)
    issue_id: uuid.UUID


class DecisionIncomingDto(DecisionDto):
    options: List[OptionIncomingDto]
    type: DecisionHierarchy = DecisionHierarchy.FOCUS


class DecisionOutgoingDto(DecisionDto):
    options: List[OptionOutgoingDto]
    type: str
